kjdoc/tour.md

```diff
@@ -1021,7 +1021,24 @@ KJ provides an advanced, cross-platform filesystem API in `kj/filesystem.h`. Fea
 * Symlinks, hard links, listing directories, recursive delete, recursive create parents, recursive copy directory, memory mapping, and unnamed temporary files are all exposed and easy to use.
 * Sparse files ("hole punching"), copy-on-write file cloning (`FICLONE`, `FICLONERANGE`), `sendfile()`-based copying, `renameat2()` atomic replacements, and more will automatically be used when available.
 
+```c++
+// NOTE: in most cases, you should construct the `Filesystem` at the entry point of your program,
+// then pass it in as a parameter wherever it's used. This allows test implementations or user-space
+// filesystems.
```

Comment:
```suggestion
// NOTE: in most cases, you should construct the `Filesystem` at the entry
// point of your program, then pass it in as a parameter wherever it's used.
// This allows dependency injection of alternative filesystem implementations
// for testing or other interesting use cases.
```

"Dependency injection" is a common term describing the point here.

I wrapped this at 80 chars since this code will go into a document / web page where width may be more constrained.